File: FSharp/Api/Data/Configuration/ApplicationUserRoleConfiguration.fs
namespace Api.Data.Configuration

open Microsoft.EntityFrameworkCore
open Microsoft.EntityFrameworkCore.Metadata.Builders
open Common.Entities

type ApplicationUserRoleConfiguration() =
  interface IEntityTypeConfiguration<ApplicationUserRole> with
    member _.Configure(builder: EntityTypeBuilder<ApplicationUserRole>) =
      builder.HasData(
        ApplicationUserRole(UserId = "userId", RoleId = "UserId"),
        ApplicationUserRole(UserId = "adminId", RoleId = "AdministratorId")
      )
      |> ignore

File: FSharp/DataContext/Data/Configuration/ApplicationUserRoleConfiguration.cs
using Common.Entities;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

namespace DataContext.Data.Configuration;

public class ApplicationUserRoleConfiguration : IEntityTypeConfiguration<ApplicationUserRole>
{
    public void Configure(EntityTypeBuilder<ApplicationUserRole> builder)
    {
        builder.HasData(new ApplicationUserRole
        {
            UserId = "userId",
            RoleId = "UserId"
        });
        builder.HasData(new ApplicationUserRole
        {
            UserId = "adminId",
            RoleId = "AdministratorId"
        });
    }
}
